frontend/src/lib/components/TaxonomicFilter/TaxonomicFilter.scss

```diff
@@ -34,6 +34,8 @@
     .taxonomic-pills {
         margin-top: 8px;
         margin-bottom: 8px;
+        padding-left: 10px;
```

Comment:
This results in a bit of an inconsistent layout for me, but not sure what'd be ideal
<img width="213" alt="Screen Shot 2022-03-24 at 12 50 37" src="https://user-images.githubusercontent.com/4550621/159910735-fd34ca91-7469-41f1-9e59-668f1bb14899.png">

Reply:
<img width="745" alt="Screen Shot 2022-03-24 at 10 00 31 AM" src="https://user-images.githubusercontent.com/254612/159970384-54515a81-a3e1-4eee-ab07-78618feec124.png">
 
Quick redline to demonstrate the intent - lists below the search are indented 8px.

Reply:
Now that the `data-management` FF has been removed and released to everyone, storybook should reflect Chris's screenshot here.